examples/example_minimal.py

```diff
@@ -0,0 +1,139 @@
+"""
+Minimal example for evolutionary regression
+===========================================
+
+Example demonstrating the use of Cartesian genetic programming for
+a simple regression task.
+"""
+
+# The docopt str is added explicitly to ensure compatibility with
+# sphinx-gallery.
+docopt_str = """
+   Usage:
+     example_minimal.py [--max-generations=<N>]
+
+   Options:
+     -h --help
+     --max-generations=<N>  Maximum number of generations [default: 300]
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.constants
+from docopt import docopt
+
+import cgp
+
+args = docopt(docopt_str)
+
+# %%
+# We first define a target function.
+
+
+def f_target(x):
+    return x[0] ** 2 + 1.0
+
+
+# %% Then we define the objective function for the evolution. It uses
+# the mean-squared error between the output of the expression
+# represented by a given individual and the target function evaluated
+# on a set of random points.
+
+
+def objective(individual):
+    if individual.fitness is not None:
+        return individual
+
+    n_function_evaluations = 1000
+
+    np.random.seed(1234)
+
+    f = individual.to_func()
+    loss = 0
+    for x in np.random.uniform(-4, 4, n_function_evaluations):
+        # the callable returned from `to_func` accepts and returns
+        # lists; accordingly we need to pack the argument and unpack
+        # the return value
+        y = f([x])[0]
+        loss += (f_target([x]) - y) ** 2
+
+    individual.fitness = -loss / n_function_evaluations
+
+    return individual
+
+
+# %%
+# Next, we set up the evolutionary search. It comprises:
+#
+# - defining the parameters for the population, the genome of individuals,
+#   and the evolutionary algorithm.
+# - creating a Population instance and instantiating the evolutionary algorithm.
+# - defining a recording callback closure for bookkeeping of the progression of the evolution.
+#
+# Finally, we call the `evolve` method to perform the evolutionary search.
+
```

Comment:
This docstring could be broken up into more pieces to make this nicer in the sphinx docs.

Reply:
hmm, it's rendered as an unordered list, isn't that what we want here?

Reply:
What I mean is: Instead of describing all the steps in one block of comment, describe them just before you execute. Let me make some concrete suggestions.